### ISSUE TITLE
JupyterHub seems not to be deployable on a Kubernetes > 1.15

### ISSUE THREAD

**Issue by Sellto** [cetic/fadi#98]
I try to deploy FADI on cluster under the 1.16 version of Kubernetes. I stuck on issue... 

## Issue
I can see the webpage of JupyterHub and i can connect to them with the default OpenLdap User/Password. But when i would like to choose a notebook, I come across this error :

 <img width="1352" alt="error" src="https://user-images.githubusercontent.com/17011743/73835305-769d0c80-480d-11ea-9901-990bf6f85453.png">

## Diagnostic

So, When i go to the logs of the hub pod, i can see :

```
[E 2020-02-05 10:47:55.598 JupyterHub gen:593] Exception in Future <Task finished coro=<BaseHandler.spawn_single_user.<locals>.finish_user_spawn() done, defined at /usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py:629> exception=TypeError("'<' not supported between instances of 'NoneType' and 'datetime.datetime'",)> after timeout
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 589, in error_callback
        future.result()
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 636, in finish_user_spawn
        await spawn_future
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/user.py", line 489, in spawn
        raise e
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/user.py", line 409, in spawn
        url = await gen.with_timeout(timedelta(seconds=spawner.start_timeout), f)
      File "/usr/local/lib/python3.6/dist-packages/kubespawner/spawner.py", line 1636, in _start
        events = self.events
      File "/usr/local/lib/python3.6/dist-packages/kubespawner/spawner.py", line 1491, in events
        for event in self.event_reflector.events:
      File "/usr/local/lib/python3.6/dist-packages/kubespawner/spawner.py", line 72, in events
        key=lambda x: x.last_timestamp,
    TypeError: '<' not supported between instances of 'NoneType' and 'datetime.datetime'
    
[E 2020-02-05 10:47:55.603 JupyterHub pages:148] Failed to spawn single-user server with form
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/pages.py", line 146, in post
        await self.spawn_single_user(user, options=options)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 737, in spawn_single_user
        status, spawner._log_name))
    tornado.web.HTTPError: HTTP 500: Internal Server Error (Spawner failed to start [status=1]. The logs for admin may contain details.)
```

The problem come from the format used by Kubernetes to send a `time` variable.

##   How to quickly solve the problem

- Stay on Kubernetes 1.15 
or
- Disable the events by modifying the `values.yaml` file of FADI. add the following configuration: 
```
  jupyterhub: 
    enabled: true
    singleuser:
      events: true
```

## How to definitively solve the problem
- Update the version of kubespawner in the jupyterHub subchart. Now the version is :
```
jupyterhub-kubespawner==0.10.1
```
## Reference 
 - [https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1444]()
 - [https://github.com/jupyterhub/kubespawner/pull/356]()

**Comment by alexnuttinck** [comment]
JupyterHub v0.9 is here: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CHANGELOG.md#090---2020-04-15

**Comment by alexnuttinck** [comment]
Resolved by the new [helm-fadi release: 0.2.4](https://github.com/cetic/helm-fadi/releases/tag/0.2.4).
I close this issue.
